File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/user/AddPrivateNetworkUseCase.java
package com.lattis.ellipse.domain.interactor.user;

import com.lattis.ellipse.data.network.model.response.AddPrivateNetworkResponse;
import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.repository.UserRepository;

import javax.inject.Inject;

import io.reactivex.Observable;

public class AddPrivateNetworkUseCase extends com.lattis.ellipse.domain.interactor.UseCase<AddPrivateNetworkResponse> {

    private UserRepository userRepository;
    private  String email;

    @Inject
    protected AddPrivateNetworkUseCase(ThreadExecutor threadExecutor,
                                       PostExecutionThread postExecutionThread,
                                       UserRepository userRepository) {
        super(threadExecutor, postExecutionThread);
        this.userRepository = userRepository;
    }
    public AddPrivateNetworkUseCase withEmail(String email) {
        this.email = email;
        return this;
    }


    @Override
    protected Observable<AddPrivateNetworkResponse> buildUseCaseObservable() {
        return userRepository.addPrivateNetworkEmail(email);
    }
}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/user/AddPrivateNetworkUseCase.kt
package com.lattis.domain.usecase.user

import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.repository.UserRepository
import com.lattis.domain.usecase.base.UseCase
import com.lattis.domain.models.PrivateNetwork
import io.reactivex.rxjava3.core.Observable
import javax.inject.Inject

class AddPrivateNetworkUseCase @Inject protected constructor(
    threadExecutor: ThreadExecutor,
    postExecutionThread: PostExecutionThread,
    private val userRepository: UserRepository
) : UseCase<List<PrivateNetwork>>(threadExecutor, postExecutionThread) {
    private var email: String? = null
    fun withEmail(email: String?): AddPrivateNetworkUseCase {
        this.email = email
        return this
    }

    override fun buildUseCaseObservable(): Observable<List<PrivateNetwork>> {
        return userRepository.addPrivateNetworkEmail(email!!)
    }

}